File: parallel_execution/sources/loop.1.c
/* 
* @@name: loop.2c
* @@type: C
* @@compilable: yes
* @@linkable: yes
* @@expect: success
* @@version: omp_5.0
*/
#include  <stdio.h>
#define N 100
int main() 
{
  float x[N], y[N];
  float a = 2.0;
  for(int i=0;i<N;i++){ x[i]=i; y[i]=0;}   // initialize
  
  #pragma omp parallel
  {
    #pragma omp loop
    for(int i = 0; i < N; ++i) y[i] = a*x[i] + y[i];
  }
  if(y[N-1] != (N-1)*2.0) printf("Error: 2*(N-1) != y[N-1]=%f",y[N-1]);
}   

File: parallel_execution/sources/loop.1.f90
! @@name:	loop.1
! @@type:	F-free
! @@operation:	run
! @@expect:	success
! @@version:	omp_5.0
program main
  integer, parameter :: N=100
  real :: x(N), y(N) 
  real :: a = 2.0e0

  x=(/ (i,i=1,N) /); y=0.0e0                    !! initialize

  !$omp parallel
    !$omp loop
       do i=1,N; y(i) = a*x(i) + y(i); enddo
  !$omp end parallel
      
  if(y(N) /= N*2.0e0) print*,"Error: 2*N /= y(N); y(N)=",y(N)
end program 
